### ISSUE TITLE
GeolocateControl: styling breaks the component

### ISSUE THREAD

**Issue by yurivangeffen** [visgl/react-map-gl#1120]
I don't think https://github.com/visgl/react-map-gl/issues/760 is fixed, or it has reappeared:
`style={{ position: 'absolute', right: 0, bottom: 0, margin: 10, marginBottom: 32}}` in version 5.2.5 creates some strange behaviour:
![Jun-02-2020 17-54-41](https://user-images.githubusercontent.com/4107720/83541667-3bc88e00-a4fa-11ea-99fe-1e372fb8d2c4.gif)

As you can see the rotating button is not outlined correctly (which makes the rotation out of center), and the location marker is not round. Like NavigationControl, it would be nice to just enclose them in a div to determine the style, right now I need the following for the shown UI components (which is just a bit awkward usage for components from the same package):
```typescript
<div>
  <GeolocateControl
    positionOptions={{ enableHighAccuracy: true }}
    trackUserLocation={true}
    showUserLocation={true}
    onViewportChange={this._handleViewportChange}
    style={{ position: 'absolute', right: 0, bottom: 0, margin: 10, marginBottom: 32}}
  />
</div>
<div style={{ position: 'absolute', right: 0, bottom: 0, margin: 10, marginBottom: 62 }}>
  <NavigationControl />
  {this.state.startpoint !== null && <Paper>
    <IconButton size='small' color='primary' aria-label='home' component='span' onClick={() => {
      this._handleViewportChange(this.state.startpoint);
    }}>
      <Home />
    </IconButton>
  </Paper>}
</div>
```

**Comment by yurivangeffen** [comment]
Now that I look at it, it seems the button width is also off by one compared to the NavigationControls?

**Comment by yurivangeffen** [comment]
The marker shape is fixed by the answer to https://github.com/visgl/react-map-gl/issues/1067, adding this to the style:
```css
.mapboxgl-user-location-dot {
  left: -100% !important;
  top: -100% !important;
}
```

**Comment by yurivangeffen** [comment]
The button rotation offset only seems to occur in zoomed out mode (100% in chrome and firefox works as expected).

**Comment by Pessimistress** [comment]
This is fixed in 6.1.0-alpha. Will be part of 5.3.0 and 6.1.0.
